cache/lru/lru.go

```diff
@@ -1,34 +1,14 @@
 package lru
 
 import (
-	"container/list"
 	"fmt"
 	"sync"
 
 	"github.com/lightninglabs/neutrino/cache"
 )
 
 // elementMap is an alias for a map from a generic interface to a list.Element.
-type elementMap[K comparable] struct {
```

Comment:
nit: Could do the list parameterization first, to avoid adding this code then removing it again.